### PR TITLE
Decprecating routed decoder

### PR DESCRIPTION
**Pull request by ejguan**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#70990 Decprecating routed decoder**

Releasing `decode` API for domains to let them implement custom `decode` DataPipe for now.

Differential Revision: [D33477620](https://our.internmc.facebook.com/intern/diff/D33477620)

### CODE REVIEW

**Comment by ejguan** [comment]
@ejguan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33477620).

File: torch/utils/data/datapipes/iter/routeddecoder.py [hunk 1]
```diff
@@ -10,7 +11,7 @@
 )
 
 
-@functional_datapipe('decode')
+@functional_datapipe('routed_decode')
```
Comment (NivekT) [inline torch/utils/data/datapipes/iter/routeddecoder.py:14]: Is there anyone using `.decode` before? Will this break anything for users?

**Review by NivekT — APPROVED** [review]
Looks like some CIs are breaking, not sure if related, but otherwise LGTM!

File: torch/utils/data/datapipes/iter/routeddecoder.py (hunk 1 above)
Comment (ejguan) [reply torch/utils/data/datapipes/iter/routeddecoder.py:14]: I am removing it because no one from domains really uses it 😢

**Comment by ejguan** [comment]
@ejguan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33477620).

**Pull request closed without merging**